packages/apple-pay-integration/src/apple-pay-button-strategy.ts

```diff
@@ -204,12 +254,23 @@ export default class ApplePayButtonStrategy implements CheckoutButtonStrategy {
                 throw new Error('Merchant validation failed');
             }
         };
+        if (this._buyNowInitializeOptions && !this._requiresShipping) {
+            applePaySession.onpaymentmethodselected = async () => {
+                await this._createBuyNowCart();
+                return this._handlePaymentMethodSelected(applePaySession);
```

Comment:
Do we need to return anything here? As far as I see, there is no return in prev implementation.

Reply:
fixed